### PR TITLE
Added summary endpoint

### PR DESCRIPTION
**Pull request by goodyduru**
Added and tested the summary endpoint with swagger comments. I had to refactor the seperate the search functionality from the extract_text function because the summary function also calls the extract_text function but doesn't want the result in the "api json" format.

### CODE REVIEW

File: api/models/extract.py [hunk 1]
```diff
@@ -640,12 +640,23 @@ def join_words(line, previous_block):
...
+    for page, content in results_dict.items():
+        for result in content:
+            results.append({
+                'page_number': page,
+                'table_segments': result,
+                'text_segments': []
+                })
+    return results
+
+def extract_text(path, text_list):
```
Comment (otseobande) [inline api/models/extract.py]: Naming a value by it's type can be cliche. `text_list` can be renamed to `texts`

File: tests/test_summary_route.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import json
+
+def test_v1_summary_response_message(client):
+    response = client.get('/v1/summary')
+    response_body = json.loads(response.data)
+
+    assert response.status_code == 200
+    assert response_body == {
+        'message': "Investor name is required to get financial summary"
```
Comment (otseobande) [inline tests/test_summary_route.py]: This this a success or an error? The message sounds like an error but the status code is 200 here

**Comment by otseobande** [comment]
I personally feel commit messages should be imperative (present-tense). It sends a clear message of the action the commit is intended to perform. Just like your branch name (add-summary-endpoint).

File: api/resources/summary.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                                        type: object
+                                        properties:
+                                            key:
+                                                type: string
+                                                example: Fund Name
+                                            value:
+                                                type: string
+                                                example: Elysian Capital II LP
+
+            404:
```
Comment (otseobande) [inline api/resources/summary.py]: Nice work with the swagger doc

File: api/models/extract.py [hunk 4]
```diff
@@ -731,5 +734,76 @@ def find_title(titles):
...
+                    if investor_value.lower() in result_key.lower():
+                        finds[investor_key] = result_value
+                        break
+        if investor_key not in finds:
+            finds[investor_key] = 'N/A'
+    return finds
+
 if __name__ == "__main__":
-    extract_text('./doc.pdf', 'South Yorkshire Pensions Authority')
+    summary('../files/doc.pdf', 'Udeme')
```
Comment (otseobande) [inline api/models/extract.py]: Is this line still needed since it looks like it was used for local testing

**Comment by otseobande** [comment]
Good job, I like how you work. Please address the comments I made.

File: api/models/extract.py (hunk 1 above)
Comment (goodyduru) [reply api/models/extract.py]: I understand, but sometimes "texts" can actually reference a section of a a page and in this context, text_list just seems clearer.

File: tests/test_summary_route.py (hunk 2 above)
Comment (goodyduru) [reply tests/test_summary_route.py]: Will fix that now.

File: api/resources/summary.py (hunk 3 above)
Comment (goodyduru) [reply api/resources/summary.py]: Thanks....

File: api/models/extract.py (hunk 1 above)
Comment (otseobande) [reply api/models/extract.py]: Okay cool

File: api/models/extract.py (hunk 4 above)
Comment (goodyduru) [reply api/models/extract.py]: Will remove that asap

File: api/models/extract.py (hunk 1 above)
Comment (goodyduru) [reply api/models/extract.py]: Will change it though.

**Pull request merged**